### PR TITLE
Refactor store: add type defs for zustand stores

### PR DESCRIPTION
**Pull request by vedantnn71**
I have:
- [x] Added type definitions for the stores and removed any/s

### CODE REVIEW

**Comment by subhoghoshX** [comment]
Thanks for working on it vedant. Unfortunately, I might be able to merge it. As I'm working on refactoring the store and the methods for a while now.

You can slim down PR to only include types i.e. everywhere I used `any`, replace that with the proper types.

**Comment by vedantnn71** [comment]
Hey @subhoghoshX, I've reverted back to previous commit and moved all stores in just one file as they were previously, now only types are added and a lot of any/s are removed in this pr.

**Comment by subhoghoshX** [comment]
There are too many changes to look at. Please, revert back the store file to its previous state. Only change the types.

**Comment by vedantnn71** [comment]
I have just:
1. Replaced explicit return statements w/ implicit returns
2. Renamed capitalized x and y to lowercase 
3. And removed any/s used while using stores in the components

Do you want to revert lowercase x and y back to capitalized ones?

**Comment by subhoghoshX** [comment]
You might have moved things up and down. The diff looks confusing.

**Comment by vedantnn71** [comment]
> You might have moved things up and down. The diff looks confusing.

now?

**Comment by subhoghoshX** [comment]
Try to sync your branch pls.

File: components/TweetCard/index.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default function TweetCard({ rootRef }: any) {
   }
   function mouseMoveHandler(e: any) {
     if (mouseDown && spaceDown) {
-      setMoveBy({ X: moveBy.X + e.movementX, Y: moveBy.Y + e.movementY });
+      setMoveBy({ x: moveBy.x + e.movementX, y: moveBy.y + e.movementY });
```
Comment (subhoghoshX) [inline components/TweetCard/index.tsx:35]:
```suggestion
      setMoveBy({ x: moveBy.X + e.movementX, y: moveBy.Y + e.movementY });
```

File: components/TweetCard/index.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ export default function TweetCard({ rootRef }: any) {
       onMouseUp={mouseUpHandler}
       onMouseMove={mouseMoveHandler}
       style={{
-        transform: `translate(${moveBy.X}px, ${moveBy.Y}px)`,
+        transform: `translate(${moveBy.x}px, ${moveBy.y}px)`,
```
Comment (subhoghoshX) [inline components/TweetCard/index.tsx:45]:
```suggestion
        transform: `translate(${moveBy.X}px, ${moveBy.Y}px)`,
```

File: store/index.ts [hunk 3]
```diff
@@ -1,143 +1,135 @@
...
-    incrementCardWidth: (c: any) => {
-      set((state: any) => ({
-        width: state.width + c,
-      }));
-    },
-    changeWidth: (c: any) => {
-      set((state: any) => ({ width: c }));
+export const useCardStore = create<CardState>((set) => ({
+  height: 332,
+  width: 672,
```
Comment (subhoghoshX) [inline store/index.ts:6]: bring back `rounded` because it doesn't work.

Comment (subhoghoshX) [inline store/index.ts:6]: bring back `rounded` because it doesn't work anymore.

**Comment by vedantnn71** [comment]
I've made the changes that you said :)

Can I add type defs for tweet store too?

File: components/TweetCard/index.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ export default function TweetCard({ rootRef }: any) {
   }
   function mouseMoveHandler(e: any) {
     if (mouseDown && spaceDown) {
-      setMoveBy({ X: moveBy.X + e.movementX, Y: moveBy.Y + e.movementY });
+      setMoveBy({ x: moveBy.X + e.movementX, y: moveBy.Y + e.movementY });
```
Comment (subhoghoshX) [inline components/TweetCard/index.tsx:35]:
```suggestion
      setMoveBy({ X: moveBy.X + e.movementX, Y: moveBy.Y + e.movementY });
```

File: store/index.ts [hunk 5]
```diff
@@ -1,143 +1,141 @@
...
-    },
-    setMoveBy: (c: any) => {
-      set((state: any) => ({ moveBy: c }));
-    },
-  };
-});
+export const usePanStore = create<PanState>((set) => ({
+  spaceDown: false,
+  mouseDown: false,
+  moveBy: { x: 0, y: 0 },
```
Comment (subhoghoshX) [inline store/index.ts:130]:
```suggestion
  moveBy: { X: 0, Y: 0 },
```

File: store/types/panState.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  mouseDown: boolean;
+  moveBy: MoveBy;
+  setSpaceDown: (spaceDown: boolean) => void;
+  setMouseDown: (mouseDown: boolean) => void;
+  setMoveBy: (moveBy: MoveBy) => void;
+}
+
+export type MoveBy = {
+  x: number;
+  y: number;
```
Comment (subhoghoshX) [inline store/types/panState.ts:12]:
```suggestion
  X: number;
  Y: number;
```

**Review by subhoghoshX — APPROVED** [review]

**Comment by subhoghoshX** [comment]
> I've made the changes that you said :)
> 
> Can I add type defs for tweet store too?

Yeah sure :)

**Comment by vedantnn71** [comment]
Done!

I think now the pr is ready to be merged

**Review by subhoghoshX — APPROVED** [review]

**Comment by subhoghoshX** [comment]
Thank you @vedantnn71.

**Comment by subhoghoshX** [comment]
Thanks, dude.

**Pull request merged**